2-Authorization-I/1-call-graph/SPA/src/index.js

```diff
@@ -1,30 +1,41 @@
-import React from "react";
-import ReactDOM from "react-dom";
+import React from 'react';
+import { createRoot } from 'react-dom/client';
+import { PublicClientApplication, EventType } from '@azure/msal-browser';
+import { App } from './App.jsx';
+import { msalConfig } from './authConfig';
+import 'bootstrap/dist/css/bootstrap.min.css';
+import './styles/index.css';
 
-import { PublicClientApplication } from "@azure/msal-browser";
-
-import { App } from "./App.jsx";
-import { msalConfig } from "./authConfig";
-
-import "bootstrap/dist/css/bootstrap.min.css";
-import "./styles/index.css";
+const container = document.getElementById('root');
+const root = createRoot(container);
 
 /**
- * MSAL should be instantiated outside of the component tree to prevent it from being re-instantiated on re-renders. 
+ * MSAL should be instantiated outside of the component tree to prevent it from being re-instantiated on re-renders.
  * For more, visit: https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/docs/getting-started.md
  */
 export const msalInstance = new PublicClientApplication(msalConfig);
 
 // Account selection logic is app dependent. Adjust as needed for different use cases.
-const accounts = msalInstance.getAllAccounts();
-
-if (accounts.length > 0) {
-  msalInstance.setActiveAccount(accounts[0]);
+if (!msalInstance.getActiveAccount() && msalInstance.getAllAccounts().length > 0) {
+    msalInstance.setActiveAccount(msalInstance.getAllAccounts()[0]);
 }
 
-ReactDOM.render(
-  <React.StrictMode>
-    <App instance={msalInstance} />
-  </React.StrictMode>,
-  document.getElementById('root')
+// Optional - This will update account state if a user signs in from another tab or window
+msalInstance.enableAccountStorageEvents();
+
+/**
+* To set an active account after the user signs in, register an event and listen to LOGIN_SUCCESS. For more,
+* visit: https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/docs/events.md
+*/
+msalInstance.addEventCallback((event) => {
+    if (event.eventType === EventType.LOGIN_SUCCESS && event.payload.account) {
+        const account = event.payload.account;
+        msalInstance.setActiveAccount(account);
+    }
+});
+
```

Comment:
It would be good to register another event for logout success and check for any more accounts in the cache